docker/Dockerfile.xenial_debug

```diff
@@ -10,17 +10,18 @@ ENV DEBIAN_FRONTEND=noninteractive
 
 RUN apt-get update && \
     apt-get install --no-install-recommends -y \
-      ca-certificates \
-      curl \
-      iptables \
-      iproute2 \
-      iputils-ping \
-      knot-dnsutils \
-      netcat \
-      tcpdump \
-      net-tools \
-      lsof \
-      linux-tools-generic \
-      sudo && apt-get upgrade -y && \
-      apt-get clean && \
-    rm -rf  /var/log/*log /var/log/apt/* /var/lib/dpkg/*-old /var/cache/debconf/*-old
+      ca-certificates=20170717~16.04.2 \
+      curl=7.47.0-1ubuntu2.13 \
+      iptables=1.6.0-2ubuntu3 \
+      iproute2=4.3.0-1ubuntu3.16.04.5 \
+      iputils-ping=3:20121221-5ubuntu2 \
+      knot-dnsutils=2.1.1-1build1 \
+      netcat=1.10-41 \
+      tcpdump=4.9.2-0ubuntu0.16.04.1 \
+      net-tools=1.60-26ubuntu1 \
+      lsof=4.89+dfsg-0.1 \
+      linux-tools-generic=4.4.0.148.156 \
+      sudo=1.8.16-0ubuntu1.6 \
+   && apt-get upgrade -y \
+   && apt-get clean \
+   && rm -rf  /var/log/*log /var/lib/apt/lists/* /var/log/apt/* /var/lib/dpkg/*-old /var/cache/debconf/*-old
```

Comment:
@howardjohn the only thing I changed in this last line was to delete the /var/lib/apt/lists/* - which deletes the apt-cache. In the best practice section for building docker images in the docker docs https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run, one of the best practices mentioned is to clean the apt-cache to reduce the image size. 

> In addition, when you clean up the apt cache by removing /var/lib/apt/lists it reduces the image size, since the apt cache is not stored in a layer.